hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/throttle/TestDynamicConfWithThroughputController.java

```diff
@@ -0,0 +1,233 @@
+
+      assertEquals(newMaxThroughputUpperBound,
+          throughputController.getMaxThroughputUpperBound(), EPSILON);
+
+      assertEquals(newMaxThroughputLowerBound,
+          throughputController.getMaxThroughputLowerBound(), EPSILON);
+
+      LOG.info("assertTrue OffPeakHour");
+
+      Thread.sleep(tunePeriod + 3000);
+      if (throughputController.getOffPeakHours().isOffPeakHour()) {
+        assertEquals(newOffpeakLimit, throughputController.getMaxThroughput(),
+            EPSILON);
+      } else {
+        assertEquals(newMaxThroughputLowerBound,
+            throughputController.getMaxThroughput(), EPSILON);
+      }
+
+    } finally {
+
+      sourceConfFile.delete();
+      FileUtils.moveFile(bakConfFile, sourceConfFile);
+
+    }
+  }
+
+  public static String txt2String(File file) {
+    StringBuilder result = new StringBuilder();
+    try {
+      BufferedReader br = new BufferedReader(new FileReader(file));
+      String s = null;
+      while ((s = br.readLine()) != null) {
+        result.append(System.lineSeparator() + s);
+      }
+      br.close();
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    return result.toString();
+  }
+
+  public void addNewConf(String newConfFile) {
+
+    DocumentBuilderFactory bdg = DocumentBuilderFactory.newInstance();
+    try {
+      DocumentBuilder builder = bdg.newDocumentBuilder();
+
+      Document dm = builder.parse(new FileInputStream(newConfFile));
+      Element root = dm.getDocumentElement();
+
+      Element propertyHigher = makeElement(dm,
+          PressureAwareCompactionThroughputController
+              .HBASE_HSTORE_COMPACTION_MAX_THROUGHPUT_HIGHER_BOUND,
+          String.valueOf(newMaxThroughputUpperBound));
+      root.appendChild(propertyHigher);
+
+      Element propertLower = makeElement(dm,
+          PressureAwareCompactionThroughputController
+              .HBASE_HSTORE_COMPACTION_MAX_THROUGHPUT_LOWER_BOUND,
+          String.valueOf(newMaxThroughputLowerBound));
+      root.appendChild(propertLower);
+
+      Element propertyStartHour = makeElement(dm,
+          CompactionConfiguration.HBASE_HSTORE_OFFPEAK_START_HOUR,
+          String.valueOf(0));
+      root.appendChild(propertyStartHour);
+
+      Element propertyEndHour =
+          makeElement(dm, CompactionConfiguration.HBASE_HSTORE_OFFPEAK_END_HOUR,
+              String.valueOf(6));
+      root.appendChild(propertyEndHour);
+
+      Element propertyOffpeakLimit = makeElement(dm,
+          PressureAwareCompactionThroughputController
+              .HBASE_HSTORE_COMPACTION_MAX_THROUGHPUT_OFFPEAK,
+          String.valueOf(newOffpeakLimit));
+      root.appendChild(propertyOffpeakLimit);
+
+      Transformer tfd = TransformerFactory.newInstance().newTransformer();
+      tfd.transform(new DOMSource(dm), new StreamResult(newConfFile));
+    } catch (ParserConfigurationException e) {
+
+    } catch (Exception e) {
+      LOG.error("addNewConf error", e);
+    }
```

Comment:
We don't want any of above Exceptions to be suppressed right? Let's remove them from here and keep them in `throws` list at method level.

Reply:
@virajjasani  Thanks for your review.I'll modify the code.